packages/node/src/methods/install-operations.ts

```diff
@@ -82,8 +104,175 @@ export async function addAppInstance(
   params.peerAddress = nodeMsg.from!;
   delete params.proposal;
   if (nodeMsg.data.proposal) {
-    await this.channels.setAppInstanceIDForProposeInstall(params);
+    await setAppInstanceIDForProposeInstall(
+      this.selfAddress,
+      this.store,
+      params
+    );
   } else {
-    await this.channels.install(params);
+    await install(this.store, params);
+  }
+}
+
+/**
+ * Creates a ProposedAppInstanceInfo to reflect the proposal received from
+ * the client.
+ * @param selfAddress
+ * @param store
+ * @param params
+ */
+async function createProposedAppInstance(
+  selfAddress: Address,
+  store: Store,
+  params: Node.ProposeInstallParams
+): Promise<string> {
+  const appInstanceId = generateUUID();
+  const channel = await getChannelFromPeerAddress(
+    selfAddress,
+    params.peerAddress,
+    store
+  );
+
+  const proposedAppInstance = new ProposedAppInstanceInfo(
+    appInstanceId,
+    params
+  );
+
+  await store.addAppInstanceProposal(channel, proposedAppInstance);
+  return appInstanceId;
+}
+
+async function install(
+  store: Store,
+  params: Node.InstallParams
+): Promise<AppInstanceInfo> {
+  if (!params.appInstanceId) {
+    return Promise.reject("No AppInstanceId specified to install");
+  }
+
+  const stateChannel = await store.getChannelFromAppInstanceID(
+    params.appInstanceId
+  );
+
+  const appInstanceInfo = await store.getProposedAppInstanceInfo(
+    params.appInstanceId
+  );
+  const appInstance: AppInstance = createAppInstanceFromAppInstanceInfo(
+    appInstanceInfo,
+    stateChannel
+  );
+  delete appInstanceInfo.initialState;
+
+  const updatedStateChannel = stateChannel.installApp(
+    appInstance,
+    appInstanceInfo.myDeposit,
+    appInstanceInfo.peerDeposit
+  );
+
+  await store.updateChannelWithAppInstanceInstallation(
+    updatedStateChannel,
+    appInstance,
+    appInstanceInfo
+  );
+
+  return appInstanceInfo;
+}
+
+/**
+ * @param appInstanceInfo The AppInstanceInfo to convert
+ * @param channel The channel the AppInstanceInfo belongs to
+ */
+function createAppInstanceFromAppInstanceInfo(
+  proposedAppInstanceInfo: ProposedAppInstanceInfo,
+  channel: StateChannel
+): AppInstance {
+  const appFunctionSigHashes = getAppFunctionSigHashes(proposedAppInstanceInfo);
+
+  const appInterface: AppInterface = {
```

Comment:
@snario are you specifically referring to return values from functions whose signature explicitly indicate the return type? Or all explicit typings for variables?

For example here: not explicitly defining the type for `appInterface` won't ensure the object has the properties you need it to have, and the error might throw downstream.

Reply:
In this case it is necessary to specify the type because it is an object literal, but there are other cases where the type can be inferred

Reply:
Right, so the former case then. Cool